File: 125-valid-palindrome/125-valid-palindrome.py
class Solution(object):
    def isPalindrome(self, s):
        """
        :type s: str
        :rtype: bool
        """
        left, right = 0, len(s)-1
        while left < right:
            if not s[left].isalnum():
                left += 1
                continue
        
            if not s[right].isalnum():
                right -= 1
                continue
        
            if s[left].lower() != s[right].lower():
                return False
        
            left += 1
            right -= 1
            
        return True

File: 125-valid-palindrome/125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        if (s.isEmpty()) {return true;}
        
        int left = 0;
        int right = s.length() - 1;
        while (left < right) {
            char lc = s.charAt(left);
            char rc = s.charAt(right);
            if (!Character.isLetterOrDigit(lc)) {
                left++;
                continue;
            }
            if (!Character.isLetterOrDigit(rc)) {
                right--;
                continue;
            }
            if (Character.toLowerCase(lc) != Character.toLowerCase(rc)) {
                return false;
            }
            left++;
            right--;
        }
        return true;
    }
}